templates/zerver/dev_login.html

```diff
@@ -57,7 +57,10 @@ <h2>Realm</h2>
 
 <script type="text/javascript">
 if (window.location.hash.substring(0, 1) === "#") {
-    document.login_form.action += window.location.hash;
+    $('input').each(function () {
+        $(this).attr('formaction',
+            $(this).attr('formaction') + '/' + window.location.hash);
+    });
```

Comment:
Can you explain this bit?

Reply:
So this is for appending hash like `#narrow/stream/49-development-help` at the end of all the formactions in the page we have.
This could of course be done in a single line as well like this:
```
$('input').attr('formaction', $('input').attr('formaction') + window.location.hash);
```
I guess I will change it to that. And the '/' can be dropped of. Not really required for dev login's but is required in the other main login page.

If you meant for me to explain how this is working, that is as below:
* So hash is basically stays on the browser side and never goes to backend. Point of doing the above it that when we submit request we are essentially kind of visiting the link. We appended the hash to that link. Any further re directions are 302's and hash won't be dropped by browser between those.
https://stackoverflow.com/questions/5283395/url-hash-is-persisting-between-redirects

Reply:
Makes sense.  Just add a brief comment, e.g. "Here we append window.location.hash, so that part of the URL can be preserved after the user logs in"